### ISSUE TITLE
bug; marker position wrong when parent has css transform: scale(1.2);

### ISSUE THREAD

**Issue by davemccoy** [jaames/iro.js#84]
Hello. I find bug. When parent/container element has css atribute transform: sclae() http://prntscr.com/pd8bs9 , when change color marker position is wrong 

https://codepen.io/dave_maccoy/pen/abogKGv

**Comment by jaames** [comment]
I'm not entirely sure why you need to use `transform: scale()` on the color picker, but that isn't going to work with iro.js

If you're trying to change the size of the color picker, you can use the `width` and `height` options: https://iro.js.org/guide.html#color-picker-options

If you need to change the size of the color picker at any point, you can use the `resize` method:
https://iro.js.org/colorPicker_api.html#methods

```js
// you specify the size you want when the color picker is created
var colorPicker = new iro.ColorPicker({
width: 300
height: 320
})

// and change it at any time with resize()
colorPicker.resize(400, 420);
```

**Comment by davemccoy** [comment]
I need to scale all block like this http://prntscr.com/pddwxy https://codepen.io/dave_maccoy/pen/abogKGv , and after transfrom marker position wrong. this is bug, why not?

**Comment by jaames** [comment]
CSS transforms like scale, rotate, etc are only visual effects, they don't change how mouse events are handled. Since the scale isnt being applied to the mouse input it treats it like the color picker isnt scaled, which is normal and can't really be fixed

**Comment by davemccoy** [comment]
I make something worflow constructor like this http://prntscr.com/pdf7x2 , so it have buttons to zoomIn/zoonOut, click on it - workflow container change it scale

**Comment by sjscotti** [comment]
I hope it isn't bad form in responding to a closed issue, but I had the same issue using iro.js colorpicker within an HTML element that I needed to have scaled using the css `scale()` transform.  I couldn't come up with a workaround outside of modifying the iro.js code itself, so explored the code and found that there were only 4 places that needed to have the scale value inserted.
(Of course, iro.js also would need to have other modifications made to include the scale factor as an input option when defining a colorpicker, and a new function added that could be called when a rescaling of the HTML element is done during the parent application execution.)

I modified the "Development Version" to change these lines in `function getSliderValueFromInput` from:
```
    if (props.layoutDirection === 'horizontal') {
      handlePos = -1 * y + handleRange + handleStart;
    } else {
      handlePos = x - handleStart;
    } // clamp handle position
```
to
```
    if (props.layoutDirection === 'horizontal') {
      handlePos = -1 * y/props.cssScale + handleRange + handleStart;	                     ////changed
    } else {
      handlePos = x/props.cssScale - handleStart;                                            ////changed
    } // clamp handle position
```

And in `function getWheelValueFromInput` I modified a couple of lines from:
```
    x = cx - x;
    y = cy - y; // Calculate the hue by converting the angle to radians
```
to:
```
    x = cx -  x/props.cssScale;									////changed
    y = cy -  y/props.cssScale; // Calculate the hue by converting the angle to radians		////changed
```

I added the css scale() value to the state as shown by `props.cssScale`  above.  The modified iro.js seems to display the cursor at (or very close to) the marker positions on the wheel and slider.  My handling of the passing of the `cssScale` value to iro.js is not as elegant as the rest of the code (which is amazing BTW!), so I would much prefer this option be added to the production code,
Thanks for reading and for such a cool color picker!
